File: produit-service/src/main/java/org/ghazoui/produitservice/services/ProduitService.java
package org.ghazoui.produitservice.services;

import org.ghazoui.produitservice.dtos.ProduitRequestDTO;
import org.ghazoui.produitservice.dtos.ProduitResponseDTO;
import java.util.List;

public interface ProduitService {

    List<ProduitResponseDTO> getAllProduits();

    ProduitResponseDTO getProduitById(Long id);

    ProduitResponseDTO createProduit(ProduitRequestDTO produitRequestDTO);

    ProduitResponseDTO updateProduit(Long id, ProduitRequestDTO produitRequestDTO);

    void deleteProduit(Long id);

    Boolean checkDisponibilite(Long id, Integer quantite);

}

File: frontend/src/app/services/produit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Produit } from '../models/produit.model';

@Injectable({
    providedIn: 'root'
})
export class ProduitService {
    private apiUrl = 'http://localhost:8888/api/produits';

    constructor(private http: HttpClient) { }

    getAllProduits(): Observable<Produit[]> {
        return this.http.get<Produit[]>(this.apiUrl);
    }

    getProduitById(id: number): Observable<Produit> {
        return this.http.get<Produit>(`${this.apiUrl}/${id}`);
    }

    createProduit(produit: Produit): Observable<Produit> {
        return this.http.post<Produit>(this.apiUrl, produit);
    }

    updateProduit(id: number, produit: Produit): Observable<Produit> {
        return this.http.put<Produit>(`${this.apiUrl}/${id}`, produit);
    }

    deleteProduit(id: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${id}`);
    }

    checkDisponibilite(id: number, quantite: number): Observable<boolean> {
        return this.http.get<boolean>(`${this.apiUrl}/${id}/disponibilite?quantite=${quantite}`);
    }
}
